pkg/polaris/azure/azure_test.go

```diff
@@ -124,29 +127,42 @@ func TestAzureSubscriptionAddAndRemove(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
-	if account.Name != testSubscription.SubscriptionName {
-		t.Fatalf("invalid name: %v", account.Name)
+	if name := account.Name; name != testSubscription.SubscriptionName {
+		t.Fatalf("invalid name: %v", name)
 	}
-	if account.NativeID != testSubscription.SubscriptionID {
-		t.Fatalf("invalid native id: %v", account.NativeID)
+	if id := account.NativeID; id != testSubscription.SubscriptionID {
+		t.Fatalf("invalid native id: %v", id)
 	}
-	if account.TenantDomain != testSubscription.TenantDomain {
-		t.Fatalf("invalid tenant domain: %v", account.TenantDomain)
+	if domain := account.TenantDomain; domain != testSubscription.TenantDomain {
+		t.Fatalf("invalid tenant domain: %v", domain)
 	}
 	if n := len(account.Features); n != 1 {
 		t.Fatalf("invalid number of features: %v", n)
 	}
-	if !account.Features[0].Equal(core.FeatureCloudNativeProtection) {
-		t.Fatalf("invalid feature name: %v", account.Features[0].Name)
+	feature, ok := account.Feature(core.FeatureCloudNativeProtection)
+	if !ok {
+		t.Fatalf("%s feature not found", core.FeatureCloudNativeProtection)
 	}
-	if regions := account.Features[0].Regions; !reflect.DeepEqual(regions, []string{"eastus2"}) {
+	if name := feature.Name; name != core.FeatureCloudNativeProtection.Name {
+		t.Fatalf("invalid feature name: %v", name)
+	}
+	if regions := feature.Regions; !reflect.DeepEqual(regions, testSubscription.CloudNativeProtection.Regions) {
```

Comment:
I'm not sure that the slice order we get back is stable, maybe we should sort to avoid flakes?